### PR TITLE
Add SERIAL type to allow auto-increment functionality

### LINKED ISSUE #17: Columns fail to autoincrement
When attempting to add a new row to a able with an auto-incrementing column, if you do not specify the value of the auto-incrementing column, the lib throws an error.
```py
import asyncio

from asyncqlio.db import DatabaseInterface
from asyncqlio.orm.schema.column import Column
from asyncqlio.orm.schema.table import table_base
from asyncqlio.orm.schema.types import Integer, String

class Test(table_base()):
    id = Column(Integer(), primary_key=True, autoincrement=True)
    name = Column(String())

async def main():
    db = DatabaseInterface('postgresql://postgres:postgres@localhost/')
    await db.connect()
    async with db.get_ddl_session() as sess:
        await sess.create_table(Test.__tablename__, *Test.iter_columns())
    db.bind_tables(Test)
    async with db.get_session() as sess:
        await sess.add(Test(name='test'))


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```
The above code raises an ```asyncpg.exceptions.NotNullViolationError```

### PR DESCRIPTION
**Pull request by BeatButton**
Fixes #17 by adding the ```SERIAL``` type, which behaves as an automatically incrementing integer.

### CODE REVIEW

File: asyncqlio/orm/schema/column.py [hunk 1]
```diff
@@ -134,6 +130,9 @@ def __init__(self, type_: 'typing.Union[md_types.ColumnType, typing.Type[md_type
         if not isinstance(self.type, md_types.ColumnType):
             # assume we need to create the "default" type
             self.type = self.type.create_default()  # type: md_types.ColumnType
+
+        self.autoincrement = isinstance(self.type, md_types.Serial)
```
Comment (SunDwarf) [inline asyncqlio/orm/schema/column.py]: We could possibly make this a property, and for sqlite3 check if it's the first column and is INTEGER PRIMARY KEY.

Comment (BeatButton) [reply asyncqlio/orm/schema/column.py]: Agree

**Comment by SunDwarf** [comment]
Also needs a BIGSERIAL and SMALLSERIAL type (inherited from Serial) for Postgres.

**Comment by SunDwarf** [comment]
Also, needs a changelog entry.

**Comment by BeatButton** [comment]
Should we be manually checking that types exist in dialects, or should we just let the error messages that will inevitably happen do that for us?
Manually checking can be error-prone and time consuming

**Comment by SunDwarf** [comment]
Let the error message happen instead.

**Pull request merged**